tests/chapter4_functional_test.py

```diff
@@ -0,0 +1,213 @@
+    """Execute a command with arguments.
+
+    Args:
+        command (string): Command to execute
+        *args (args list): Arguments to be forwarded to the
+
+    Returns:
+        bool: True if command execution did not throw any errors, False otherwise
+    """
+    try:
+        subprocess.run(args=[command, *args], check=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL)
+        return True
+    except:
+        return False
+
+def run_cmake(*args):
+    """Run CMake with given args
+
+    Returns:
+        bool: True if command execution did not throw any errors, False otherwise
+    """
+    return run_command(CMAKE_COMMAND, *args)
+
+def run_cpack(*args):
+    """Run CPack with given args
+
+    Returns:
+        bool: True if command execution did not throw any errors, False otherwise
+    """
+    return run_command(CPACK_COMMAND, *args)
+
+def test_prerequisites():
+    """
+    Check overall prerequisites for the tests.
+    
+    If one of the checks fail, the testing cannot proceed any further.
+    """
+    assert run_cmake("--version"), "CMake is not present!"
+    assert run_cpack("--version"), "CPack is not present!"
+    # assert run_command("rpmbuild", "--version"), "rpmbuild is not present!"
+    # assert run_command("dpkg", "--version"), "DPKG is not present!"
+    assert run_command("tar", "--version"), "TAR is not present!"
+
+class TestChapter4():
+    """
+    Tests for chapter 4 CMake example content.
+    
+    This code validates that can be configured, built, installed without any issues. In addition,
+    the test code validates installation has installed the expected files (and not installed the unexpected ones) 
+    and tries to run the executables.
+    """
+    temporary_build_root_dir=tempfile.TemporaryDirectory("build")
+    temporary_install_root_dir=tempfile.TemporaryDirectory("install")
+    temporary_build_root=temporary_build_root_dir.name
+    temporary_install_root=temporary_install_root_dir.name
+    source_directory=str(get_chapter4_root_directory().absolute())
+    
+    def test_configure(self):
+        """Check if project can be configured
+        """
+        assert run_cmake("-S", self.source_directory, "-B", self.temporary_build_root)
+
+    def test_build(self):
+        """Check if project can be built
+        """
+        assert run_cmake("--build", self.temporary_build_root, *CMAKE_COMMAND_MULTICONFIG_ARGS,  "--parallel", str(os.cpu_count()))
+
+    def test_install(self):
+        """Check if project can be installed
+        """
+        assert run_cmake("--install", self.temporary_build_root, *CMAKE_COMMAND_MULTICONFIG_ARGS, "--prefix", self.temporary_install_root)
+
+    def test_install_check_files_exists(self):
+        """Check if install step has installed the all files that install() commands intend to install
+        """
+        # Example 1
+        assert file_exists(f"{self.temporary_install_root}/bin/ch4_ex01_executable{PLATFORM_EXECUTABLE_EXTENSION}")
```

Comment:
The test is ok, but this seems to depend on the `test_install(self)` or am I mistaken?